src/libraries/System.Text.Json/Common/JsonSimpleNamingPolicy.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Buffers;
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSimpleNamingPolicy : JsonNamingPolicy
```

Comment:
"Simple" is fairly ambiguous, would recommend a name suggesting we're inserting separator chars, e.g.
```suggestion
    internal sealed class JsonSeparatorNamingPolicy : JsonNamingPolicy
```

Reply:
Originally, not in the pull request, I had `JsonNamingPolicy<TTransform> where TTransform : struct, ITransform`. That allowed me to have a generic implementation covering all cases, including the camel. I guess, it's too late to alter the camel case policy which currently only changes the first letter without affecting the rest? In other libraries, all policies take an input, split it into words and join them with or without mutations.

Reply:
What would such an `ITransform` do? Is it different to the `JsonNamingPolicy` as an abstraction?

Reply:
That interface had one method to append a passed word to the resulting buffer.

```
interface ITransform
{
    bool TryWrite(ReadOnlySpan<char> word, Span<char> result);
}
```

No logic on buffer resize, just doing upper/lower case transform on the `word` and applying it to the buffer, plus a separator if needed. In case if the camel case policy follows the same logic as for the proposed in that PR policies, then it will lower case the first word and capitalize others. That's a bit slower than the current implementation affecting the first char only, but then all policies will have the same transform rules on the input. Surely, that's a breaking change.

The idea of `TTransform` is to avoid a virtual call which happens if `JsonSeparatorNamingPolicy` would be an abstract type with an abstract `TryWrite` method. So since `TTransform` is a struct the call even can be inlined and do direct `ToLowerInvariant` or `ToUpperInvariant` invocations with a separator prefixing that. Plus no heap allocations due to in place, on stack, allocation of the state.

I haven't published that version of my package and has no public commits, but can restore that and post here if needed.

Reply:
It's needed only in case when camel case shall extract words. For example, `SomeName_Underscored` with the proposed policies will be translated to `some_name_underscored` as an example, but the already existing camel case gives `someName_Underscored`.

Without a change in the current camel case that `TTransform` isn't needed surely.